crates/project_panel/src/project_panel.rs

```diff
@@ -2572,6 +2583,45 @@ impl ProjectPanel {
         }
     }
 
+    fn collect_selected_files(&self, cx: &Context<Self>) -> Vec<(Entity<Worktree>, PathBuf)> {
```

Comment:
The name is somewhat confusing, and we only use this as a paths for diff, so let's make a proper interface for that?

```suggestion
    fn file_to_diff_abs_paths(&self, cx: &Context<Self>) -> Option<(PathBuf, PathBuf)> {
```

Inside, we could do all that

```

        let (worktree1, path1) = &selected_files[0];
        let (worktree2, path2) = &selected_files[1];
        let abs_path1 = worktree1.read(cx).absolutize(path1).ok();
        let abs_path2 = worktree2.read(cx).absolutize(path2).ok();
```

selection and normalization (is this the only reason we drag around the worktrees? Cannot we absolutize paths on the spot??)

Reply:
You're right. I didn't think about the sequence.
I want to use IndexMap/IndexSet instead of Vec for better performance. But no one seems to use this data structure. I wonder if this is appropriate?

Reply:
We do not have a way to select many entries (e.g. cmd/ctrl-a does not work), so I suspect a regular `Vec` is fine.

Reply:
We can select many entries with Shift.

Reply:
Indeed, thank you for correcting.

I think that is still relatively hard to use to select so many entries to start seeing issues: for that, one needs to scroll past multiple thousands of entries and click on the top again — possible, but how probable it is?

So, while we aim the "regular" operations here and need an ordered sequence of entries selected, we should be ok and I can solve whatever edge case perf fall-outs caused by this later.